### DISCUSSION TITLE
Session file affects disk space.

### DISCUSSION THREAD

**Discussion by MarcosDLR** [open-wa/wa-automate-nodejs#2572] [Q&A]
Hello.

I know this is not a bug, I would just like information regarding the following:

When I connect the phone number a folder with the name of the session is created, this folder grows as time goes by and disk space is affected. Is there any way to prevent this ?

**Comment by smashah** [comment]
I don't mind making some sort of script that trims the fat from the folder however, people will need to collate the list of resources that are safe to delete.

Most of the space is taken by the `Default` folder which itself is mostly `Cache`.

![CleanShot 2022-03-15 at 14 09 12](https://user-images.githubusercontent.com/5903821/158396408-b707d3b3-a2a7-495f-bdbb-143c7cefb691.png)

![CleanShot 2022-03-15 at 14 10 50](https://user-images.githubusercontent.com/5903821/158396450-ced2655e-262d-479e-b738-8d234bfc5c4f.png)

The cache includes things like the loading animation and the js/css/html resources for WA WEB which are served via the service worker. The service worker is bypassed anyways so the cache can theoretically be cleared.

**Comment by smashah** [comment]
This line forces the cache to stay enabled (default).

https://github.com/open-wa/wa-automate-nodejs/blob/ae6c5df7f85a2f2d1728e1e699539bb6fbf1a41b/src/controllers/browser.ts#L79

Will test by removing this

**Comment by MarcosDLR** [comment]
Thank you very much.

**Comment by smashah** [comment]
@MarcosDLR MD sessions reauthenticate just fine without forcing the cache to be enabled.

I will remove the if statement in the above mentioned line. This should disable the cache. Not sure if this will prevent the storage increase over time but lets try.

**Comment by smashah** [comment]
@github-actions run

<details>
<summary>⚡ Release!  ⚡</summary>

```js
(async () => {
function exec(cmd) {
  console.log(execSync(cmd).toString());
}
//set the version type
process.env.VERS = "patch"

// Config
const gitUserEmail = "github-actions[bot]@users.noreply.github.com";
const gitUserName = "github-actions[bot]";

exec(`echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc`);
exec(`git config --global user.email "${gitUserEmail}"`);
exec(`git config --global user.name "${gitUserName}"`);
exec(`npm i -D`);
exec(`npm run release-ci $VERS`);

// types only package
exec('npm config set //registry.npmjs.org/:_authToken $NPM_TOKEN')
exec(`cd ./types-only && npm version $VERS && npm run build && npm publish && cd ..`);
exec(`git commit -a -m 'updated types-only package'`);
exec(`git push --force`);

//comment on the issue
var result = execSync(`npx auto-changelog -o ./tempchangelog.txt --commit-limit false --template ./compact-keepachangelog.hbs --stdout`).toString();

    await postComment(result);

//create changelog image
exec(`npm run release-image`);
exec(`git commit -a -m 'updated release-image'`);
exec(`git push --force`);
  })();
```
</details>

**Comment by smashah** [comment]
# Changelog

## 🚀 Release [4.32.8](https://github.com/open-wa/wa-automate-nodejs/compare/4.32.7...4.32.8) (2022-03-15)

- Release 4.32.8 [`265ae2a`](https://github.com/open-wa/wa-automate-nodejs/commit/265ae2a2ab8d5c4ac9233414e1410903c1c0d3e9)
- 🔧  dont force the cache to be enabled on MD mode #2566 [`07e1e4e`](https://github.com/open-wa/wa-automate-nodejs/commit/07e1e4e3f86d5fdd23f0ab2bf05effcbb072cdf2)
- updated types-only package [`ae6c5df`](https://github.com/open-wa/wa-automate-nodejs/commit/ae6c5df7f85a2f2d1728e1e699539bb6fbf1a41b)

**Comment by MarcosDLR** [comment]
Hello again, sorry for the inconvenience.

With the latest update the cache is resolved but the \Default\IndexedDB\https_web.whatsapp.com_0.indexeddb.leveldb folder grows exponentially. This creates .ldb and .log files.

**Comment by smashah** [comment]
@MarcosDLR can you delete the folder, re scan and track the growth over time with exact numbers please.

**Comment by MarcosDLR** [comment]
After 10 minutes the file grew 1,449 kb, this was with 90 messages received, but the file grows even without sending or receiving messages. The file grows depending on the type of message.

After several days this becomes a considerable consumption of space depending on the flow of messages.

![image](https://user-images.githubusercontent.com/47016656/158596827-d5ce6eb9-ddc6-429f-9dac-48d74d6aa4e5.png)
